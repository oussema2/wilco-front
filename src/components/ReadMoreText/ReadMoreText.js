import React from 'react';
import PropTypes from 'prop-types';
import ReadMore from '@fawazahmed/react-native-read-more';
import { View } from 'react-native';
import { styles } from './ReadMoreText.styles';

const ReadMoreText = ( {
	testID, text, onPress, style
} ) => (
	<View testID={testID}>
		<ReadMore
			style={style}
			seeMoreText=" ... more"
			seeLessText="  See less"
			seeMoreStyle={styles.seeMore}
			seeLessStyle={styles.seeLess}
			ellipsis=""
			numberOfLines={2}
			onPress={onPress}
		>
			{text}
		</ReadMore>
	</View>
);

ReadMoreText.propTypes = {
	testID: PropTypes.string,
	text: PropTypes.string,
	onPress: PropTypes.func,
	style: PropTypes.instanceOf( Object )
};

ReadMoreText.defaultProps = {
	testID: 'read-more-testID',
	text: '',
	onPress: () => {},
	style: {}
};

export default ReadMoreText;
