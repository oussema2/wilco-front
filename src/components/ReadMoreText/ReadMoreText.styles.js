import { StyleSheet } from 'react-native';
import { palette } from '../../Theme';

export const styles = StyleSheet.create( {
	seeMore: {
		color: palette.grayscale.shutterGrey
	},
	seeLess: {
		color: palette.grayscale.shutterGrey
	}
} );
