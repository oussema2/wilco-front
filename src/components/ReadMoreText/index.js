export { default as ReadMoreText } from './ReadMoreText';
