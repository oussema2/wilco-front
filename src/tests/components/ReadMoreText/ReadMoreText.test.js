import * as React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import ReadMoreText from '../../../components/ReadMoreText/ReadMoreText';

describe( 'ReadMoreText', () => {
	describe( 'ReadMoreText component with the default props', () => {
		it( 'renders the ReadMoreText component correctly', async () => {
			const testID = 'read-more-testID';
			const component = render( <ReadMoreText /> );

			expect( component.queryByTestId( testID ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'ReadMoreText component without default props', () => {
		it( 'renders the ReadMoreText component correctly', async () => {
			const testID = 'test-testID';
			const text = 'Text for test ReadMoreText component.';
			const onPress = jest.fn();
			const component = render( <ReadMoreText testID={testID} text={text} onPress={onPress} /> );

			fireEvent.press( component.queryByText( text ) );
			expect( onPress ).toHaveBeenCalledTimes( 1 );
			expect( component.queryByTestId( testID ) ).not.toBeNull();
			expect( text ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );
} );
