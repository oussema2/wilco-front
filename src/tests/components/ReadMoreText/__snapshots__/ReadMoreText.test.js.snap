// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMoreText ReadMoreText component with the default props renders the ReadMoreText component correctly 1`] = `
<View
  testID="read-more-testID"
>
  <View
    style={
      Object {
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={2}
      onPress={[Function]}
      style={Object {}}
      text=""
    >
      
    </Text>
  </View>
</View>
`;

exports[`ReadMoreText ReadMoreText component without default props renders the ReadMoreText component correctly 1`] = `
<View
  testID="test-testID"
>
  <View
    style={
      Object {
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={2}
      onPress={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      style={Object {}}
      text=""
    >
      Text for test ReadMoreText component.
    </Text>
  </View>
</View>
`;
